queue_processors/autoreduction_processor/reduction_service.py

```diff
@@ -0,0 +1,210 @@
+        except AssertionError as ex:
+            raise DatafileError(f"Problem reading datafile: {path}") from ex
+
+
+class ReductionScript:
+    """
+    Encapsulates the loading and running of a reduction script
+    """
+
+    def __init__(self, instrument):
+        self.script_path = Path(MISC["scripts_directory"] % instrument) / "reduce.py"
+        self.skipped_runs = []
+        self.script = None
+
+    def load(self):
+        """
+        Loads the reduction script as a module and stores any run numbers that are to be skipped
+        by the script
+        """
+        spec = spec_from_file_location("reducescript", self.script_path)
+        self.script = module_from_spec(spec)
+        spec.loader.exec_module(self.script)
+        try:
+            self.skipped_runs = self.script.SKIP_RUNS
+        except:  # pylint:disable=bare-except
+            pass
+
+    def run(self, input_file, output_dir):
+        """
+        Runs the reduction script on the given input file and outputs to the given
+        ReductionReturn and returns the return value of the main function of the script.
+        :param input_file: (Datafile) Input datafile
+        :param output_dir: (ReductionDirectory) Directory to output to
+        :return:
+        """
+        LOGGER.info("Running reduction script: %s", self.script_path)
+        with TimeOut(MISC["script_timeout"]):
+            return self.script.main(input_file=str(input_file.path),
+                                    output_dir=str(output_dir.path))
+
+
+# pylint:disable=too-many-arguments; We will remove the log_Stream once we look at logging in ppa
+# more closely
+def reduce(reduction_dir, temp_dir, datafile, script, run_number):
+    """
+    Performs a reduction on the given datafile using the given script, outputting to the given
+    output directory
+    :param reduction_dir: (ReductionDirectory) The final directory to output to
+    :param temp_dir: (TemporaryReductionDirectory) Where the reduction initially outputs to
+    :param datafile: (Datafile) The datafile to perform the reduction on
+    :param script: (ReductionScript) The Script used to reduce the data
+    :param run_number: (String) The run number of this reduction
+    :return (StringIO): The log stream of the reduction script
+    """
+    reduction_dir.create()
+    LOGGER.info("-------------------------------------------------------")
+    LOGGER.info("Temporary result directory: %s", temp_dir.path)
+    LOGGER.info("Final Result directory: %s", reduction_dir.path)
+    LOGGER.info("Temporary log dir: %s", temp_dir.log_path)
+    LOGGER.info("Final log dir: %s", reduction_dir.log_path)
+    LOGGER.info("Datafile: %s", datafile.path)
+    LOGGER.info("Reduction script: %s", script.script_path)
+    LOGGER.info("-------------------------------------------------------")
+    LOGGER.info("Starting reduction...")
+
+
+    script.load()
+    if run_number in script.skipped_runs:
```

Comment:
Hi @keiranjprice101 I looked and there is no reference to `skipped_runs` in any of our reduce.py and reduce_vars.py files, and neither on help page on webapp. Hence please delete this if statement. Further I am not sure there is any point in the SkippedRunException. Unless you see reason for keeping it could you delete this exception also please